test/commands/init_test.js

```diff
@@ -10,258 +10,190 @@
-  }
-
-  test('registers default template generators', () => {
-    let registeredGens = {};
-    let registerSpy = sandbox.spy(function (gen, name) {
-      registeredGens[name] = gen;
+  test('calls ', () => {
+    let registeredGenerators = initCommand.env.getGeneratorsMeta();
+    assert.deepEqual(registeredGenerators, {
+      'polymer-init-element:app': {
+        resolved: 'unknown',
+        namespace: 'polymer-init-element:app',
+      },
+      'polymer-init-application:app': {
+        resolved: 'unknown',
+        namespace: 'polymer-init-application:app',
+      },
+      'polymer-init-shop:app': {
+        resolved: 'unknown',
+        namespace: 'polymer-init-shop:app',
+      },
+      'polymer-init-app-drawer-template:app': {
+        resolved: 'unknown',
+        namespace: 'polymer-init-app-drawer-template:app',
+      },
     });
-    yeomanEnvStub.prototype.registerStub = registerSpy;
-
-    yeomanEnvStub.prototype.getGeneratorsMeta.returns(registeredGens);
-    stubInquirer('polymer-init-element:app');
-
-    return new InitCommand().run({}, {})
-      .then(() => {
-        assert.isTrue(registerSpy.calledWith(
-          match.same(elementGeneratorStub.ElementGenerator),
-          'polymer-init-element:app'
-        ));
+  });
 
-        assert.isTrue(registerSpy.calledWith(
-          match.same(applicationGeneratorStub.ApplicationGenerator),
-          'polymer-init-application:app'
-        ));
+  test('prompts with a list of all registered generators', (done) => {
+    let promptStub = sandbox.stub(inquirer, 'prompt').returns(new Promise(function() {}));
+    cli.run().then(() => {
+      assert.isTrue(promptStub.calledOnce);
+      assert.isTrue(promptStub.calledWith([
+        {
+          "type":"list",
+          "name":"generatorName",
+          "message":"Which starter template would you like to use?",
+          "choices":[
+            {
+              "name":"element: \u001b[2mA blank element template\u001b[22m",
+              "value":"polymer-init-element:app",
+              "short":"element",
+            },
+            {
+              "name":"application: \u001b[2mA blank application template\u001b[22m",
+              "value":"polymer-init-application:app",
+              "short":"application",
+            },
+            {
+              "name":"shop: \u001b[2mThe \"Shop\" Progressive Web App demo\u001b[22m",
+              "value":"polymer-init-shop:app",
+              "short":"shop",
+            },
+            {
+              "name":"app-drawer-template: \u001b[2mA starter application template, with navigation and \"PRPL pattern\" loading\u001b[22m",
+              "value":"polymer-init-app-drawer-template:app",
+              "short":"app-drawer-template",
+            },
+          ],
+        },
+      ]));
+      done();
+    });
+  });
 
-        assert.isTrue(registerSpy.calledWith(
+  test('includes user-provided generators in the list when properly installed/registered', () => {
```

Comment:
fi we assume we've tested the ability of the command to actually perform the prompting, then testing user-installed generators is once of the things that would be a lot easier by just testing the generator list functionality directly.